sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialImdsLiveTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.Testing;
+using Azure.Identity.Tests.Mock;
+using Azure.Security.KeyVault.Secrets;
+using NUnit.Framework;
+
+namespace Azure.Identity.Tests
+{
+    public class ManagedIdentityCredentialImdsLiveTests : RecordedTestBase
+    {
+        public ManagedIdentityCredentialImdsLiveTests(bool isAsync) : base(isAsync)
+        {
+            Sanitizer = new IdentityRecordedTestSanitizer();
+        }
+
+        [SetUp]
+        public void ResetManagedIdenityClient()
+        {
+            typeof(ManagedIdentityClient).GetField("s_msiType", BindingFlags.NonPublic | BindingFlags.Static).SetValue(null, 0);
```

Comment:
Can we add an internal method for this? This piece of reflection gets copied around a lot.

Reply:
The need for this has been removed since I've made the the discovery instance based.